README.rst

```diff
@@ -686,13 +686,17 @@ variable ``FUN``:
         is given, then all the ``bin`` passed will be prepended with this
         ``path``. ``bin``\ s can be passed as multiple arguments or as one
         list.
-``mapfiles(src, dst, file[, ...][, append_suffix=True])``
+``mapfiles(src, dst, [file[, ...]][, append_suffix=True])``
         A very simple function that just returns a list with
         ``{src}/{file}={dst}/{file}{VAR.suffix}`` for each ``file`` passed.
-        ``file``\ s can be passed as multiple arguments or as one list. A named
-        argument ``append_suffix`` can be passed at the end to control whether
-        ``VAR.suffix`` is appended to each destination file. ``append_suffix``
-        defaults to ``True`` if not given.
+        ``file``\ s can be passed as multiple arguments or as one list. If no
+        ``file``\ s are passed, a simple ``{src}/={dst}/`` mapping is returned
+        which results in the directory structure under ``src`` (including all
+        contained files) being replicated as-is from the local filesystem into
+        ``dst`` within the package. A named argument ``append_suffix`` can be
+        passed at the end to control whether ``VAR.suffix`` is appended to each
+        destination file. ``append_suffix`` defaults to ``True`` if not given,
+        and is only applicable when at least one ``file`` is passed.
```

Comment:
I'm sorry but I'm still not convinced :confused: 

For me adding a wrapper that does basically nothing is making things more obscure. At the end this is calling `fpm`, and I feel by abusing of `FUN.mapfiles()` we are making users of this wrapper distance themselves from the original format, that on itself has its own subtleties (like is not the same if you use a trailing `/` or not).

The `mapfiles()` function (or more correctly the original `mapbins()` function existed only to deal with the annoyance of having to include the suffix for a bunch of files, but when you remove the suffix from the scene, for me there is no place for this function. Maybe except (again) if you need to map a bunch of individual files, in which case it saves you from repetition (or more complex python list comprehensions). But if you don't need to add suffixes and all your files are in the same directory, unless there are other files in that directory that you don't want to include, I don't even see a case for `mapfiles()` as you can map the whole directory as `src/=dst`.

So again, `mapfiles()` was supposed to remove boilerplate, not just to add a layer of obscurity over a very clear and simple `src/=dst` string.

To finish I want to stress that I agree that the code looks more regular and nicer when using `mapfiles()` everywhere. Like is *graphically* nicer, but I think in terms of what is actually going on there, is way more obscure. So, how I am supposed to know that `FUN.mapfiles(VAR.bindir, '/usr/bin', app1, app2)` and `FUN.mapfiles('configs', '/etc/configs', config1, config2)` are adding suffixes and `FUN.mapfiles('helpers', '/etc/helpers')` does not (BTW, I really wonder if you intended to add suffixes to configuration files). So you have two almost identical pieces of code doing totally different things. That for me is a false (and dangerous) *readability improvement*.